service/src/main/kotlin/fi/espoo/evaka/application/ApplicationStateService.kt

```diff
@@ -557,19 +559,23 @@ class ApplicationStateService(
             throw BadRequest("Expected status to be one of [${statuses.joinToString(separator = ", ")}] but was ${application.status}")
     }
 
+    fun canApplyOnPreschoolTerm(tx: Database.Read, preferredStartDate: LocalDate): Boolean {
+        val term = tx.getActivePreschoolTermAt(preferredStartDate) ?: return false
+        val applicationsAccepted = ClosedPeriod(term.applicationPeriod.start, term.extendedTerm.end)
```

Comment:
Would it be better to model this logic inside `PreschoolTerm` as now every user of these periods has to know about this?